tests/ctl/api/test_seed.py

```diff
@@ -436,3 +444,193 @@ async def test_load_orginizations(loguru_caplog, async_session, monkeypatch):
 
     assert "INSERTED 1" in loguru_caplog.text
     assert f"SKIPPED {current_orgs}" in loguru_caplog.text
+
+
+@pytest.mark.integration
+class TestLoadSampleResources:
+    """Tests related to load_sample_resources"""
+
+    SAMPLE_ENV_VARS = {
+        # Include test secrets for Postgres and Stripe, only
+        "FIDES_DEPLOY__CONNECTORS__POSTGRES__HOST": "test-var-expansion",
+        "FIDES_DEPLOY__CONNECTORS__POSTGRES__PORT": "9090",
+        "FIDES_DEPLOY__CONNECTORS__POSTGRES__DBNAME": "test-var-db",
+        "FIDES_DEPLOY__CONNECTORS__POSTGRES__USERNAME": "test-var-user",
+        "FIDES_DEPLOY__CONNECTORS__POSTGRES__PASSWORD": "test-var-password",
+        "FIDES_DEPLOY__CONNECTORS__STRIPE__DOMAIN": "test-stripe-domain",
+        "FIDES_DEPLOY__CONNECTORS__STRIPE__API_KEY": "test-stripe-api-key",
+    }
+
+    @patch.dict(os.environ, SAMPLE_ENV_VARS, clear=True)
+    async def test_load_samples(
+        self,
+        async_session: AsyncSession,
+    ) -> None:
+        """
+        Test that we can load the sample resources, connections, and upsert those
+        into the database. See the other tests in this class for more detailed
+        assertions - this one just ensures the e2e result is what we expect: a
+        database full of sample data!
+        """
+
+        # Load the sample resources & connections
+        await seed.load_samples(async_session)
+
+        async with async_session.begin():
+            # Check the results are as expected!
+            systems = (await async_session.execute(select(System))).scalars().all()
+            datasets = (await async_session.execute(select(Dataset))).scalars().all()
+            policies = (await async_session.execute(select(PolicyCtl))).scalars().all()
+            connections = (
+                (await async_session.execute(select(ConnectionConfig))).scalars().all()
+            )
+            dataset_configs = (
+                (await async_session.execute(select(DatasetConfig))).scalars().all()
+            )
+            assert len(systems) == 4
+            assert len(datasets) == 3
+            assert len(policies) == 1
+            assert len(connections) == 2
+            assert len(dataset_configs) == 2
```

Comment:
For those keeping track at home:
* 4 systems (defined in sample_resources/sample_systems.yml)
* 3 datasets (2 defined in sample_resources/ for Postgres & Mongo, 1 defined in the Stripe SaaS connector template)
* 1 policy (define in sample_resources/sample_policy.yml)
* 2 connections (for this test, only Postgres & Stripe have their secrets defined)
* 2 dataset configs (Postgres & Stripe)